### PR TITLE
FF trust cert

### LINKED ISSUE #22019: [HTTPS] Add instructions for trusting the developer certificate on Linux
Within [this](https://docs.microsoft.com/en-us/aspnet/core/security/enforcing-ssl?view=aspnetcore-5.0&tabs=visual-studio#trust-https-certificate-on-linux) section

## Make a section for trusting the certificate in edge/chrome on Linux
* You need to have installed `libnss3-tools` on your distribution.
* There should be a folder `$HOME/.pki/nssdb` on the machine.
* Export the certificate with `dotnet dev-certs https -ep /usr/local/share/ca-certificates/aspnet/https.crt --format PEM`
* Run the following commands and restart the browser for it to take effect
  ```
  certutil -d sql:$HOME/.pki/nssdb -A -t "P,," -n localhost -i /usr/local/share/ca-certificates/aspnet/https.crt
  certutil -d sql:$HOME/.pki/nssdb -A -t "C,," -n localhost -i /usr/local/share/ca-certificates/aspnet/https.crt
  ```
## Make a section for trusting the certificate in Firefox
* Export the certificate with `dotnet dev-certs https -ep /usr/local/share/ca-certificates/aspnet/https.crt --format PEM`
* Create file `/usr/lib/firefox/distribution/policies.json` with the following contents
  ```json
  {
      "policies": {
          "Certificates": {
              "Install": [
                  "/usr/local/share/ca-certificates/aspnet/https.crt"
              ]
          }
      }
  }
  ```

## Make a section for Ubuntu to make dotnet trust the certificate (for service-to-service communication).
* **Requires openssl 1.1.1h and higher**. For instructions on how to update openssl find a guide for your OS.
* Run the following two commands:
  ```
  sudo dotnet dev-certs https -ep /usr/local/share/ca-certificates/aspnet/https.crt --format PEM
  sudo update-ca-certificates
  ```

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: c7ddfb2f-5dea-5e31-2a27-02a4dc16104c
* Version Independent ID: 12ca0b9a-2638-a4ce-c45f-d4579c83dafb
* Content: [Enforce HTTPS in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/security/enforcing-ssl?view=aspnetcore-5.0&tabs=visual-studio)
* Content Source: [aspnetcore/security/enforcing-ssl.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/security/enforcing-ssl.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-security**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### LINKED ISSUE #22020: [HTTPS] Add instructions to trust the HTTPS certificate on Firefox
Add a section to trust the HTTPS certificate within Firefox on **Windows and Mac OS** within [this section](https://docs.microsoft.com/en-us/aspnet/core/security/enforcing-ssl?view=aspnetcore-5.0&tabs=visual-studio#trust-the-aspnet-core-https-development-certificate-on-windows-and-macos)

* Create a policy file at `C:\Program Files\Mozilla Firefox\distribution\policies.json` on windows or at `Firefox.app/Contents/Resources/distribution` on Mac with the following contents:
  ```
  {
    "policies": {
      "Certificates": {
        "ImportEnterpriseRoots": true
      }
    }
  }
  ```

This makes Firefox trust certificates from the trusted certificates in the Windows certificate store. If you don't want this, there are alternative ways to configure it described in the firefox documentation [here](https://github.com/mozilla/policy-templates/blob/master/README.md)

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: c7ddfb2f-5dea-5e31-2a27-02a4dc16104c
* Version Independent ID: 12ca0b9a-2638-a4ce-c45f-d4579c83dafb
* Content: [Enforce HTTPS in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/security/enforcing-ssl?view=aspnetcore-5.0&tabs=visual-studio#trust-https-certificate-on-linux)
* Content Source: [aspnetcore/security/enforcing-ssl.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/security/enforcing-ssl.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-security**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### LINKED ISSUE #22021: [HTTPS] Correct instructions to use the HTTPS development certificate on wsl
In [this section](https://docs.microsoft.com/en-us/aspnet/core/security/enforcing-ssl?view=aspnetcore-5.0&tabs=visual-studio#trust-https-certificate-from-windows-subsystem-for-linux)

Replace the existing instructions with the following:
* Export the developer certificate to a file on **Windows**
   ```console
   dotnet dev-certs https --ep C:\<<path-to-folder>>\aspnetcore.pfx -p <<password>>
   ```
* Import the exported certificate on the WSL instance
  ```console
  dotnet dev-certs https --clean --import /mnt/c/<<path-to-folder>>/aspnetcore.pfx -p <<password>>
  ```

This is a one time operation per certificate and per wsl distro compared to having to export the certificate over and over (if you update/regenerate the certificate on windows you might need to run it again)

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: c7ddfb2f-5dea-5e31-2a27-02a4dc16104c
* Version Independent ID: 12ca0b9a-2638-a4ce-c45f-d4579c83dafb
* Content: [Enforce HTTPS in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/security/enforcing-ssl?view=aspnetcore-5.0&tabs=visual-studio#trust-https-certificate-on-linux)
* Content Source: [aspnetcore/security/enforcing-ssl.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/security/enforcing-ssl.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-security**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Fixes #22020
Fixes #22021
Fixes #22019

[Internal review URL](https://review.docs.microsoft.com/en-us/aspnet/core/security/enforcing-ssl?view=aspnetcore-5.0&branch=pr-en-us-22064&tabs=visual-studio#trust-the-aspnet-core-https-development-certificate-on-windows-and-macos)

### CODE REVIEW

File: aspnetcore/security/enforcing-ssl.md [hunk 1]
```diff
@@ -343,38 +343,75 @@ The following command provides help on the `dev-certs` tool:
...
 
   ```
-    ASPNETCORE_Kestrel__Certificates__Default__Password="<cryptic-password>" 
-    ASPNETCORE_Kestrel__Certificates__Default__Path=/mnt/c/Users/user-name/.aspnet/https/aspnetapp.pfx
-    dotnet watch run
+  dotnet dev-certs https --clean --import /mnt/c/<<path-to-folder>>/aspnetcore.pfx -p <<password>>
   ```
 
-  The preceding command sets the environment variables so Linux uses the Windows trusted certificate.
+  The preceding approach is a one time operation per certificate and per WSL distribution. It's easier than exporting the certificate over and over. If you update or regenerate the certificate on windows, you might need to run the preceding commands again.
```
Comment (Rick-Anderson) [inline aspnetcore/security/enforcing-ssl.md:414]: > It's easier than exporting the certificate over and over.

When would you need to do it over? Can't we remove this. 
***is a one time operation per certificate and per WSL distribution.*** should be all we need to say

File: aspnetcore/security/enforcing-ssl.md [hunk 2]
```diff
@@ -343,6 +343,43 @@ The following command provides help on the `dev-certs` tool:
...
+    "Certificates": {
+      "ImportEnterpriseRoots": true
+    }
+  }
+}
+```
+
+The preceding policy file makes Firefox trust certificates from the trusted certificates in the Windows certificate store. The next section provides an alternative approach for the Firefox browser.
+
+### Firefox SEC_ERROR_INADEQUATE_KEY_USAGE certificate error
```
Comment (javiercn) [inline aspnetcore/security/enforcing-ssl.md:369]: I don't remember this bit being required.

The json policy above is doing precisely this.

File: aspnetcore/security/enforcing-ssl.md [hunk 3]
```diff
@@ -351,30 +388,79 @@ See [this GitHub issue](https://github.com/dotnet/AspNetCore.Docs/issues/6199).
...
+Establishing trust is browser specific. The following sections provide instructions for the Chromium browsers Edge and Chrome and for Firefox.
+
+### Trust HTTPS certificate on Linux using Edge or Chrome
+
+  * Install the `libnss3-tools` for your distribution.
+  * Create or verify the `$HOME/.pki/nssdb` folder exists on the machine.
+  * Export the certificate with the following command:
+  
+    ```cli
+    dotnet dev-certs https -ep /usr/local/share/ca-certificates/aspnet/https.crt --format PEM
```
Comment (javiercn) [inline aspnetcore/security/enforcing-ssl.md:400]: Worth mentioning that the path here is specific for Ubuntu, but that they can choose a different path or the path for the CA-store on their distro.

File: aspnetcore/security/enforcing-ssl.md [hunk 4]
```diff
@@ -351,30 +388,79 @@ See [this GitHub issue](https://github.com/dotnet/AspNetCore.Docs/issues/6199).
...
+          "Certificates": {
+              "Install": [
+                  "/usr/local/share/ca-certificates/aspnet/https.crt"
+              ]
+          }
+      }
+  }
+  ```
+
+## Ubuntu trust the certificate for service-to-service communication
```
Comment (javiercn) [inline aspnetcore/security/enforcing-ssl.md:434]: It might be worth putting this section first and making the path comment for other distros here

**Review by javiercn — APPROVED** [review]
looks great!

File: aspnetcore/security/enforcing-ssl.md (hunk 4 above)
Comment (Rick-Anderson) [reply aspnetcore/security/enforcing-ssl.md]: Can you verify I put this is the right place?

File: aspnetcore/security/enforcing-ssl.md (hunk 2 above)
Comment (Rick-Anderson) [reply aspnetcore/security/enforcing-ssl.md]: The browser approach is much simpler and works on all platforms. Can I put the browser approach first? See how I've done it. I can remove the browser approach if you think that's best.

**Review by javiercn — APPROVED** [review]

**Pull request merged**